feat: integrate expense chart with backend

nabilmuafa opened the pull request:
### Summary
This PR integrates the outflow/expense chart with the backend services, replacing placeholder data with real-time values and aligning the component with the client's design expectations.

### Key Changes
* **Connected Chart to Backend**: The outflow chart now fetches real data via a custom data hook that communicates with the backend service.

* **New Outcome Data Hook**: Introduced a reusable hook to retrieve and manage outflow data, streamlining data flow for the chart component.

* **UI & Design Enhancements**: Updated the chart's layout and styling to closely match the client's design specifications, with added responsiveness for various screen sizes.

* **Full Test Coverage**: All newly added logic and components are covered with tests, ensuring 100% code coverage.

### Known Issues

* Implementation is still a little bit cluttered; some functions used in the outcome chart should be a reusable one from another function (for example, the outcome data hook is very similar to the income data code), yet it is redundantly rewritten. Needs to be refactored.
* Chart pre-made component is not efficiently used as there is still some need to directly use external libraries about chart.

Any feedbacks would be appreciated. Thank you!

venedictchen commented:
Hi Nabil, Great work on implementing the outflow chart. Yes, I think we can refactor in the future so the income and outcome chart can be reusable one to another because they are just different slightly.

nabilmuafa commented:
Hi @alifbintang1, thank you for pointing that out! They're actually not duplicated tests as they're in a different `describe()` block. The ones I added are intentionally for the outcome data hook. But now that you've mentioned it, some tests in those are still using the income data hook. The tests are similar with the income data hook, so I copy pasted it but I forgot to adjust the hooks to reference outcome data hook. Fixed it in the recent commit. Please check that out!

fadhilmuh commented:
Hi Nabil, good work on the PR!

One suggestion — Avoid Silent Failures (Error Handling)
You're logging the error to the console, which is fine for debugging, but i think it's not ideal. Was this intentional or not? Let me know.

nabilmuafa commented:
> Hi Nabil, good work on the PR!
> 
> One suggestion — Avoid Silent Failures (Error Handling) You're logging the error to the console, which is fine for debugging, but i think it's not ideal. Was this intentional or not? Let me know.
> 
> in frontend/src/hooks/useOutcomeData.js
> 
> ```
> } catch (err) {
>         console.error('Error fetching outcome data:', err);
>         setError(err.message);
>       } finally {
>         setIsLoading(false);
>       }
> ```
> 
> Instead of just logging, consider returning a meaningful error message that the UI can display — or even use a toast/snackbar system to inform the user when something goes wrong.

That is intentional, and it is already handled in the UI to display the error. If the error happens, the `error` state will be updated with the error message then returned. Then in `dashboard.jsx`, if the error state isn't null (meaning that there's an error), the chart will not be rendered and instead the error message will.

venedictchen commented on frontend/src/components/outflow-chart.jsx:
```diff
@@ -23,14 +16,54 @@ import {
     color: '#800000',
   },
 };
 
```
Hi Nabil, Great work on the implementation. We will considering the refactor later.

venedictchen reviewed (commented):
Hi Nabil, Great job. We will considering the refactor later.
